lib/ui/natives.dart

```diff
@@ -35,6 +35,8 @@ void _setupHooks() {
   }());
 }
 
+void saveCompilationTrace(String filePath) native 'SaveCompilationTrace';
```

Comment:
Lets do some basic sanity checking on the arguments in Dart itself before making the native call. Stuff like null or zero length file paths.

Reply:
FYI, this argument is already validated for null by Dart_NewStringFromCString() and shows nice stack trace. And empty value is caught by ::open :)